lab-carolina/app/entry.js

```diff
@@ -0,0 +1,39 @@
+'use strict';
+
+require('./scss/main.scss');
+
+const cowsay = require('cowsay-browser');
+const angular = require('angular');
+
+const demoApp = angular.module('demoApp', []);
+
+demoApp.controller('CowsayController', ['$log', CowsayController]);
+
+function CowsayController($log){
+  this.title = 'Make us speak!';
+  this.history = [];
+
+  cowsay.list((err, list) => {
+    this.list = list;
+    this.currentCow = 'kitty';
+  });
+
+  this.updateCow = function(input){
+    return '\n' + cowsay.think({text: input || 'What am I thinking about', f: this.currentCow});
+  };
+
+  this.currText = function(input){
+    this.prevText = this.updateCow(input);
+    this.history.push(this.prevText);
+  };
+
+  this.undo = function(){
+    this.history.pop();
```

Comment:
This is broken logic here, given that you're popping off your history array twice. It 'works' but isn't correct. Give it some thought, and play around with it in the browser to see if you can refactor to the correct logic.